icu4c/source/common/uloc.cpp

```diff
@@ -457,73 +457,21 @@ NULL
 typedef struct CanonicalizationMap {
     const char *id;          /* input ID */
     const char *canonicalID; /* canonicalized output ID */
-    const char *keyword;     /* keyword, or NULL if none */
-    const char *value;       /* keyword value, or NULL if kw==NULL */
 } CanonicalizationMap;
 
 /**
  * A map to canonicalize locale IDs.  This handles a variety of
  * different semantic kinds of transformations.
  */
 static const CanonicalizationMap CANONICALIZE_MAP[] = {
-    { "",               "en_US_POSIX", NULL, NULL }, /* .NET name */
-    { "c",              "en_US_POSIX", NULL, NULL }, /* POSIX name */
```

Comment:
This is the cause of regression https://unicode-org.atlassian.net/browse/ICU-20575 -  "c" should NOT have been removed here.

Reply:
> ICU uloc_canonicalize() will no longer map "C" or "POSIX" to "en_US_POSIX", but uprv_getPOSIXIDForCategory() still does so.

two different comparisons were conflated. the strcmp("C") was to check whether the posix layer was returning real data or not. The mapping still needed to be done in uloc_canonicalize()